core/lib/commonProps.ts

```diff
@@ -0,0 +1,6 @@
+export interface AllWidgetsCommonPropsAndState {
```

Comment:
I think that just `WidgetsCommonPropsAndState` as a name is enough ;)

Reply:
Thank you for your suggestion. I have renamed the interface.